onnxruntime/core/providers/webnn/builders/impl/split_op_builder.cc

```diff
@@ -0,0 +1,162 @@
+  std::vector<int64_t> input_shape;
+  ORT_RETURN_IF_NOT(GetShape(*input_defs[0], input_shape, logger), "Cannot get shape");
+  const auto rank = input_shape.size();
+  emscripten::val options = emscripten::val::object();
+
+  NodeAttrHelper helper(node);
+  int64_t axis = helper.Get("axis", 0);
+  axis = HandleNegativeAxis(axis, rank);
+  options.set("axis", static_cast<int32_t>(axis));
+
+  if (input_defs.size() == 2) {
+    // Inputs contains optional 'split' input
+    std::vector<int64_t> split;
+    const auto& initializers(model_builder.GetInitializerTensors());
+    const auto& split_tensor = *initializers.at(input_defs[1]->Name());
+    ORT_RETURN_IF_NOT(GetShapeByTensor(split_tensor, split, logger), "Cannot get split.");
+    std::vector<int32_t> splits;
+    std::transform(split.cbegin(), split.cend(),
+                   std::back_inserter(splits),
+                   [](int64_t dim) -> int32_t { return SafeInt<int32_t>(dim); });
+    output_array = model_builder.GetBuilder().call<emscripten::val>("split", input, emscripten::val::array(splits), options);
+    ORT_RETURN_IF_NOT(output_array["length"].as<int32_t>() == static_cast<int32_t>(split.size()), "The size of outputs must be equal to the size of 'split' input.");
+  } else {
+    if (helper.HasAttr("num_outputs")) {
+      const int64_t num_outputs = helper.Get("num_outputs", 1);
+      if (input_shape[axis] % num_outputs == 0) {
+        // The 'num_outputs' evenly divide the dim value at 'axis' specified.
+        output_array = model_builder.GetBuilder().call<emscripten::val>("split", input, static_cast<int32_t>(num_outputs), options);
+      } else {
+        std::vector<int64_t> mapping_split;
+        mapping_split.insert(mapping_split.begin(), num_outputs - 1, input_shape[axis] / num_outputs);
+        mapping_split.insert(mapping_split.end(), input_shape[axis] % num_outputs);
+        std::vector<int32_t> converted_splits;
+        std::transform(mapping_split.cbegin(), mapping_split.cend(),
+                       std::back_inserter(converted_splits),
+                       [](int64_t dim) -> int32_t { return SafeInt<int32_t>(dim); });
+        output_array = model_builder.GetBuilder().call<emscripten::val>("split", input, emscripten::val::array(converted_splits), options);
+      }
+      ORT_RETURN_IF_NOT(output_array["length"].as<int32_t>() == static_cast<int32_t>(num_outputs), "The size of outputs must be equal to 'num_outputs'.");
+    } else {
+      // w/o 'split' input for opset 13
+      // split: Optional length of each output. Values should be >= 0.Sum of the values must be equal to the dim value at 'axis' specified.
+      output_array = model_builder.GetBuilder().call<emscripten::val>("split", input, static_cast<int32_t>(1), options);
+      ORT_RETURN_IF_NOT(output_array["length"].as<int32_t>() == static_cast<int32_t>(1), "The size of outputs must be equal to 1.");
+    }
+  }
+  for (int64_t i = 0; i < output_array["length"].as<int32_t>(); i++) {
```

Comment:
```suggestion
  for (int64_t i = 0, count = output_array["length"].as<int32_t>(); i < count; i++) {
```

(minor) Tis nice to assign the count to a temporary, which allows easy inspection in the debugger and avoids re-evualing the keyed lookup and as each time.